shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -74,386 +296,493 @@ impl SimpleRedisCache {
+    }
 
-            // TODO: Translate the redis_responses into a cassandra result
-            *cass_request = Message::from_frame(Frame::Cassandra(CassandraFrame {
-                version: Version::V4,
-                operation: CassandraOperation::Result(CassandraResult::Void),
-                stream_id: cass_request.stream_id().unwrap(),
-                tracing_id: None,
-                warnings: vec![],
-            }));
+    /// Clears the cache for the entire table
+    /// TODO make this drop only the specified keys not the entire cache
+    fn clear_table_cache(&self) -> Option<Message> {
+        Some(Message::from_frame(Frame::Redis(RedisFrame::BulkString(
+            "FLUSHDB".into(),
+        ))))
+    }
+
+    /// clear the cache for the single row specified by the redis_key
+    fn clear_row_cache(
+        &mut self,
+        statement: &CassandraStatement,
+        table_cache_schema: &TableCacheSchema,
+    ) -> Option<Message> {
+        // TODO is it possible to return the future and process in parallel?
+
+        if let Ok((redis_key, _hash_key)) = build_redis_key_from_cql3(statement, table_cache_schema)
+        {
+            let commands_buffer: Vec<RedisFrame> = vec![
+                RedisFrame::BulkString("DEL".into()),
+                RedisFrame::BulkString(redis_key),
+            ];
+            Some(Message::from_frame(Frame::Redis(RedisFrame::Array(
+                commands_buffer,
+            ))))
+        } else {
+            None
         }
-        Ok(messages_cass_request)
     }
-}
 
-// TODO: We don't need to do it this way and allocate another struct
-struct ValueHelper(Value);
-
-impl ValueHelper {
-    fn as_bytes(&self) -> &[u8] {
-        match &self.0 {
-            Value::Number(v, false) => v.as_bytes(),
-            Value::SingleQuotedString(v) => v.as_bytes(),
-            Value::NationalStringLiteral(v) => v.as_bytes(),
-            Value::HexStringLiteral(v) => v.as_bytes(),
-            Value::Boolean(v) => {
-                if *v {
-                    &TRUE
+    /// calls the next transform and process the result for caching.
+    async fn execute_upstream_and_process_result<'a>(
+        &mut self,
+        message_wrapper: Wrapper<'a>,
+    ) -> ChainResponse {
+        let mut orig_messages = message_wrapper.messages.clone();
+        let orig_cql: Option<&mut CQL> = orig_messages
+            .iter_mut()
+            .filter_map(|message| {
+                if let Some(Frame::Cassandra(CassandraFrame {
+                    operation: CassandraOperation::Query { query, .. },
+                    ..
+                })) = message.frame()
+                {
+                    Some(query)
                 } else {
-                    &FALSE
+                    None
+                }
+            })
+            .next();
+        let result_messages = &mut message_wrapper.call_next_transform().await?;
```

Comment:
We can avoid copying all the message by making this an owned value instead of a reference
![image](https://user-images.githubusercontent.com/5120858/168508040-c9e01bb4-051f-4d51-94d6-8f1756d02d76.png)